pkg/services/object/internal/client/client.go

```diff
@@ -237,97 +319,196 @@ func (x PayloadRangeRes) PayloadRange() []byte {
 // Returns:
 //  error of type *object.SplitInfoError if object if raw flag is set and requested object is virtual;
 //  object.ErrAlreadyRemoved error if requested object is marked to be removed.
-func PayloadRange(prm PayloadRangePrm) (res PayloadRangeRes, err error) {
-	res.cliRes, err = prm.cli.ObjectPayloadRangeData(prm.ctx, &prm.cliPrm, prm.opts...)
-	if err == nil {
-		// pull out an error from status
-		err = apistatus.ErrFromStatus(res.cliRes.Status())
+func PayloadRange(prm PayloadRangePrm) (*PayloadRangeRes, error) {
+	if prm.local {
+		prm.cliPrm.MarkLocal()
+	}
+
+	if prm.tokenSession != nil {
+		prm.cliPrm.WithinSession(*prm.tokenSession)
+	}
+
+	if prm.tokenBearer != nil {
+		prm.cliPrm.WithBearerToken(*prm.tokenBearer)
+	}
+
+	rdr, err := prm.cli.ObjectRangeInit(prm.ctx, prm.cliPrm)
+	if err != nil {
+		return nil, fmt.Errorf("init payload reading: %w", err)
+	}
+
+	data, err := io.ReadAll(rdr)
+	if err != nil {
+		return nil, fmt.Errorf("read payload: %w", err)
```

Comment:
Maybe allocate `make([]byte, prm.Length()` like we do in other places? Less reallocations.